### DISCUSSION TITLE
Environment Variables in Edge Functions

### DISCUSSION THREAD

**Discussion by ijaspreetsinghh** [supabase/supabase#8075] [Questions]
I am trying to introduce env secrets in local edge functions.

![image](https://user-images.githubusercontent.com/44422128/182774421-deb157a7-9a79-41bd-965c-a926df9184a7.png)

Already have the local env file create with ```echo "MY_NAME=Yoda" >> ./supabase/.env.local``` as per documentation.

![image](https://user-images.githubusercontent.com/44422128/182774642-846f3fac-8b4d-44d6-869a-e874ec2f0db0.png)

env file contains above value, but still, unable to read the value using ```console.log(Deno.env.get('MY_NAME'))```.

if run by ```supabase functions serve add-project```, ```console.log(Deno.env.get('MY_NAME'))``` prints ***undefined***, and if run via command ```supabase functions serve add-project --env-file ./supabase/.env.local```, it throws an error of : ***Error: Can't separate key from value***.
 
![image](https://user-images.githubusercontent.com/44422128/182775957-30eaee5f-9838-4291-af80-03634417fa20.png)

![image](https://user-images.githubusercontent.com/44422128/182775045-f08794ec-9e03-4b80-b83e-d75bc1e95420.png)

![image](https://user-images.githubusercontent.com/44422128/182774918-8d8dfff6-7bac-41b9-bc43-87965f20a708.png)

**Comment by mojogogo** [comment]
I'm getting the exact same error as documented above. Did you ever resolve this? Does anybody have any suggestions?

**Comment by cohlar** [comment]
For me, adding ` --env-file ./supabase/.env.local` works. Without it, the environment variables aren't picked up automatically either.
